File: gait/waveform_data/UFO_Waveform_Plotting.R
#Plotting script to plot individual participant waveforms and group means +/- sd
#Updated: 4/25/21


##A sad attempt at plotting multiple waveforms on a plot from Sunday 4/25/21
#Pre-Fatigue SS
setwd("/Users/rossbrancati/Desktop/UFO/Gait/Data/Participant_Mean_Waveforms/pre_fatigue_ss")
file_list <- list.files(path = "/Users/rossbrancati/Desktop/UFO/Gait/Data/Participant_Mean_Waveforms/pre_fatigue_ss")

pre_ss_all <- data.frame()

for (i in 1:length(file_list)){
    temp_file <- read.csv(file_list[i], header = FALSE)
    pre_ss_all <- rbind(pre_ss_all, temp_file)
}

ggplot(data=pre_ss_all) + 
  geom_line(aes(x=c(0:706),y=V4))

##Plotting Mean +/- SD 
pre_ss_walk = read.csv('/Users/rossbrancati/Desktop/UFO/Gait/waveform_data/young/matlab_exports/means_and_stdevs/pre_fatigue_ss.csv')
post_ss_walk = read.csv('/Users/rossbrancati/Desktop/UFO/Gait/waveform_data/young/matlab_exports/means_and_stdevs/post_fatigue_ss.csv')
pre_fast_walk = read.csv('/Users/rossbrancati/Desktop/UFO/Gait/waveform_data/young/matlab_exports/means_and_stdevs/pre_fatigue_fast.csv')
post_fast_walk = read.csv('/Users/rossbrancati/Desktop/UFO/Gait/waveform_data/young/matlab_exports/means_and_stdevs/post_fatigue_fast.csv')

#Assigning variables to data
current_pre_speed = pre_fast_walk
current_post_speed = post_fast_walk

#import libraries
library("ggplot2")

##################
###Joint Angles###
######Stance######
##################
ggplot() + 
  geom_ribbon(data=current_pre_speed, aes(x=c(0:100), ymin = (hip_angle_x_s-hip_angle_x_s_stdev), ymax = (hip_angle_x_s+hip_angle_x_s_stdev)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_pre_speed, aes(x=c(0:100),y=hip_angle_x_s), color=rgb(136,28,28,maxColorValue=255)) +
  geom_ribbon(data=current_post_speed, aes(x=c(0:100), ymin = (hip_angle_x_s-hip_angle_x_s_stdev), ymax = (hip_angle_x_s+hip_angle_x_s_stdev)), fill = "black", alpha=0.2) +
  geom_line(data=current_post_speed, aes(x=c(0:100),y=hip_angle_x_s), color="black") +
  scale_x_continuous(name="Percent Stance (%)", expand = c(0,0), limits = c(1,100))+
  ylab ("Angle (Degrees)") + 
  ggtitle("Hip Flexion Angle - Stance") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, face="bold", size=20), axis.title = element_text(size = 20),
        axis.text.y = element_text(size = 16), axis.text.x = element_text(size = 16), plot.margin = margin(10, 20, 10, 10)) +
  geom_hline(yintercept=0) 
  
ggplot() + 
  geom_ribbon(data=current_pre_speed, aes(x=c(0:100), ymin = (knee_angle_x_s-knee_angle_x_s_stdev), ymax = (knee_angle_x_s+knee_angle_x_s_stdev)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_pre_speed, aes(x=c(0:100),y=knee_angle_x_s), color=rgb(136,28,28,maxColorValue=255)) +
  geom_ribbon(data=current_post_speed, aes(x=c(0:100), ymin = (knee_angle_x_s-knee_angle_x_s_stdev), ymax = (knee_angle_x_s+knee_angle_x_s_stdev)), fill = "black", alpha=0.2) +
  geom_line(data=current_post_speed, aes(x=c(0:100),y=knee_angle_x_s), color="black") +
  scale_x_continuous(name="Percent Stance (%)", expand = c(0,0), limits = c(1,100))+
  ylab ("Angle (Degrees)") + 
  ggtitle("Knee Flexion Angle - Stance") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, face="bold", size=20), axis.title = element_text(size = 20),
        axis.text.y = element_text(size = 16), axis.text.x = element_text(size = 16), plot.margin = margin(10, 20, 10, 10)) +
  geom_hline(yintercept=0)

ggplot() + 
  geom_ribbon(data=current_pre_speed, aes(x=c(0:100), ymin = (ankle_angle_x_s-ankle_angle_x_s_stdev), ymax = (ankle_angle_x_s+knee_angle_x_s_stdev)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_pre_speed, aes(x=c(0:100),y=ankle_angle_x_s), color=rgb(136,28,28,maxColorValue=255)) +
  geom_ribbon(data=current_post_speed, aes(x=c(0:100), ymin = (ankle_angle_x_s-ankle_angle_x_s_stdev), ymax = (ankle_angle_x_s+knee_angle_x_s_stdev)), fill = "black", alpha=0.2) +
  geom_line(data=current_post_speed, aes(x=c(0:100),y=ankle_angle_x_s), color="black") +
  scale_x_continuous(name="Percent Stance (%)", expand = c(0,0), limits = c(1,100))+
  ylab ("Angle (Degrees)") + 
  ggtitle("Ankle Flexion Angle - Stance") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, face="bold", size=20), axis.title = element_text(size = 20),
        axis.text.y = element_text(size = 16), axis.text.x = element_text(size = 16), plot.margin = margin(10, 20, 10, 10)) +
  geom_hline(yintercept=0) + 
  annotate("text", x = 11, y = 20, label = "Dorsiflexion (+)") +
  annotate("text", x = 11, y = -15, label = "Plantarflexion (-)")


##################
###Joint Angles###
###Gait Cycle#####
##################
ggplot() + 
  geom_ribbon(data=current_pre_speed, aes(x=c(0:100), ymin = (hip_angle_x_f-hip_angle_x_f_stdev), ymax = (hip_angle_x_f+hip_angle_x_f_stdev)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_pre_speed, aes(x=c(0:100),y=hip_angle_x_f), color=rgb(136,28,28,maxColorValue=255)) +
  geom_ribbon(data=current_post_speed, aes(x=c(0:100), ymin = (hip_angle_x_f-hip_angle_x_f_stdev), ymax = (hip_angle_x_f+hip_angle_x_f_stdev)), fill = "black", alpha=0.2) +
  geom_line(data=current_post_speed, aes(x=c(0:100),y=hip_angle_x_f), color="black") +
  scale_x_continuous(name="Percent Gait Cycle (%)", expand = c(0,0), limits = c(1,100))+
  ylab ("Angle (Degrees)") + 
  ggtitle("Hip Flexion Angle - Gait Cycle") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, face="bold", size=20), axis.title = element_text(size = 20),
        axis.text.y = element_text(size = 16), axis.text.x = element_text(size = 16), plot.margin = margin(10, 20, 10, 10)) +
  geom_hline(yintercept=0)

ggplot() + 
  geom_ribbon(data=current_pre_speed, aes(x=c(0:100), ymin = (knee_angle_x_f-knee_angle_x_f_stdev), ymax = (knee_angle_x_f+knee_angle_x_f_stdev)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_pre_speed, aes(x=c(0:100),y=knee_angle_x_f), color=rgb(136,28,28,maxColorValue=255)) +
  geom_ribbon(data=current_post_speed, aes(x=c(0:100), ymin = (knee_angle_x_f-knee_angle_x_f_stdev), ymax = (knee_angle_x_f+knee_angle_x_f_stdev)), fill = "black", alpha=0.2) +
  geom_line(data=current_post_speed, aes(x=c(0:100),y=knee_angle_x_f), color="black") +
  scale_x_continuous(name="Percent Gait Cycle (%)", expand = c(0,0), limits = c(1,100))+
  ylab ("Angle (Degrees)") + 
  ggtitle("Knee Flexion Angle - Gait Cycle") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, face="bold", size=20), axis.title = element_text(size = 20),
        axis.text.y = element_text(size = 16), axis.text.x = element_text(size = 16), plot.margin = margin(10, 20, 10, 10)) +
  geom_hline(yintercept=0)

ggplot() + 
  geom_ribbon(data=current_pre_speed, aes(x=c(0:100), ymin = (ankle_angle_x_f-ankle_angle_x_f_stdev), ymax = (ankle_angle_x_f+knee_angle_x_f_stdev)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_pre_speed, aes(x=c(0:100),y=ankle_angle_x_f), color=rgb(136,28,28,maxColorValue=255)) +
  geom_ribbon(data=current_post_speed, aes(x=c(0:100), ymin = (ankle_angle_x_f-ankle_angle_x_f_stdev), ymax = (ankle_angle_x_f+knee_angle_x_f_stdev)), fill = "black", alpha=0.2) +
  geom_line(data=current_post_speed, aes(x=c(0:100),y=ankle_angle_x_f), color="black") +
  scale_x_continuous(name="Percent Gait Cycle (%)", expand = c(0,0), limits = c(1,100))+
  ylab ("Angle (Degrees)") + 
  ggtitle("Ankle Flexion Angle - Gait Cycle") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, face="bold", size=20), axis.title = element_text(size = 20),
        axis.text.y = element_text(size = 16), axis.text.x = element_text(size = 16), plot.margin = margin(10, 20, 10, 10)) +
  geom_hline(yintercept=0) + 
  annotate("text", x = 11, y = 20, label = "Dorsiflexion (+)") +
  annotate("text", x = 11, y = -15, label = "Plantarflexion (-)")

###################
###Joint Moments###
###################
ggplot() + 
  geom_ribbon(data=current_pre_speed, aes(x=c(0:100), ymin = (hip_mom_x-hip_mom_x_stdev), ymax = (hip_mom_x+hip_mom_x_stdev)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_pre_speed, aes(x=c(0:100),y=hip_mom_x), color=rgb(136,28,28,maxColorValue=255)) +
  geom_ribbon(data=current_post_speed, aes(x=c(0:100), ymin = (hip_mom_x-hip_mom_x_stdev), ymax = (hip_mom_x+hip_mom_x_stdev)), fill = "black", alpha=0.2) +
  geom_line(data=current_post_speed, aes(x=c(0:100),y=hip_mom_x), color="black") +
  scale_x_continuous(name="Percent Stance (%)", expand = c(0,0), limits = c(1,100))+
  ylab ("Moment (%BW*Height)\nEXT(-)/FLX(+)") + 
  ggtitle("Hip Flexion/Extension Moment") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, face="bold", size=20), axis.title = element_text(size = 12),
        axis.text.y = element_text(size = 16), axis.text.x = element_text(size = 16), plot.margin = margin(10, 20, 10, 10)) +
  geom_hline(yintercept=0)

ggplot() + 
  geom_ribbon(data=current_pre_speed, aes(x=c(0:100), ymin = (knee_mom_x-knee_mom_x_stdev), ymax = (knee_mom_x+knee_mom_x_stdev)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_pre_speed, aes(x=c(0:100),y=knee_mom_x), color=rgb(136,28,28,maxColorValue=255)) +
  geom_ribbon(data=current_post_speed, aes(x=c(0:100), ymin = (knee_mom_x-knee_mom_x_stdev), ymax = (knee_mom_x+knee_mom_x_stdev)), fill = "black", alpha=0.2) +
  geom_line(data=current_post_speed, aes(x=c(0:100),y=knee_mom_x), color="black") +
  scale_x_continuous(name="Percent Stance (%)", expand = c(0,0), limits = c(1,100))+
  ylab ("Moment (%BW*Height)\nFLX(-)/EXT(+)") + 
  ggtitle("Knee Flexion/Extension Moment") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, face="bold", size=20), axis.title = element_text(size = 12),
        axis.text.y = element_text(size = 16), axis.text.x = element_text(size = 16), plot.margin = margin(10, 20, 10, 10)) +
  geom_hline(yintercept=0)

ggplot() + 
  geom_ribbon(data=current_pre_speed, aes(x=c(0:100), ymin = (ankle_mom_x-ankle_mom_x_stdev), ymax = (ankle_mom_x+ankle_mom_x_stdev)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_pre_speed, aes(x=c(0:100),y=ankle_mom_x), color=rgb(136,28,28,maxColorValue=255)) +
  geom_ribbon(data=current_post_speed, aes(x=c(0:100), ymin = (ankle_mom_x-ankle_mom_x_stdev), ymax = (ankle_mom_x+ankle_mom_x_stdev)), fill = "black", alpha=0.2) +
  geom_line(data=current_post_speed, aes(x=c(0:100),y=ankle_mom_x), color="black") +
  scale_x_continuous(name="Percent Stance (%)", expand = c(0,0), limits = c(1,100))+
  ylab ("Moment (%BW*Height)\nDF(-)/PF(+)") + 
  ggtitle("Ankle Plantarflexion/Dorsiflexion Moment") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5, face="bold", size=16), axis.title = element_text(size = 12),
        axis.text.y = element_text(size = 16), axis.text.x = element_text(size = 16), plot.margin = margin(10, 20, 10, 10)) +
  geom_hline(yintercept=0)

File: gait/waveform_data/UFO_waveform_plotting.m
%This function plots each participants average for each condition on one
%plot. End result is four figures with hip, knee, and ankle sagittal plane 
%joint angles (stance and full gait cycle), moments, and powers (one
%figure) per condition).
%
%Updated 8/2/21 by Ross

function UFO_waveform_plotting(base_file_path,group)

    mean_dir = dir([base_file_path,group,'/matlab_exports/Participant_Mean_Waveforms']);
    mean_dir(strncmp({mean_dir.name},'.',1))=[];

    participant_cell_array = cell(0,0);
    participant_cell_array{1,1} = ['post_fatigue_fast'];
    participant_cell_array{1,2} = ['post_fatigue_ss'];
    participant_cell_array{1,3} = ['pre_fatigue_fast'];
    participant_cell_array{1,4} = ['pre_fatigue_ss'];


    for i = 1:length(mean_dir)

        current_folder = mean_dir(i,1);
        current_struct = dir(strcat(current_folder.folder,'/',current_folder.name));
        current_struct(strncmp({current_struct.name},'.',1))=[];
        current_struct(contains({current_struct.name},'DS_Store'))=[];
        
        sum_mat = zeros(101,24);
        
        for p = 1:length(current_struct)

            current_file_path = strcat(current_struct(p).folder,'/',current_struct(p).name);
            current_file = importdata(current_file_path);
            sum_mat = sum_mat + current_file;
            participant_cell_array{p+1,i} = current_file;

        end

    end

    [participant_cell_array_r, participant_cell_array_c] = size(participant_cell_array);
    x = linspace(0,100,101)';

    for cond = 1:4


        for sub_id = 2:participant_cell_array_r

            figure(cond)
            %%HIP%%
            %Hip flexion angle stance
            hold on;
            subplot(3,4,1);
            plot(x,(participant_cell_array{sub_id,cond}(:,3)));
            title({'Hip Flexion Angle','Stance Phase'});
            ylabel('Joint angle (degrees)');
            xlabel('Percent Stance (%)');

            %Hip flexion angle GC
            hold on;
            subplot(3,4,2);
            plot(x,(participant_cell_array{sub_id,cond}(:,9)));
            title({'Hip Flexion Angle','Gait Cycle'});
            ylabel('Joint angle (degrees)');
            xlabel('Percent Gait Cycle (%)');

            %Hip flexion moment
            hold on;
            subplot(3,4,3);
            plot(x,(participant_cell_array{sub_id,cond}(:,16)));
            title({'Hip Joint Moment','Stance Phase'});
            ylabel('Joint Moment (%BW*Height)');
            xlabel('Percent Stance (%)');

            %Hip Joint Power
            hold on;
            subplot(3,4,4);
            plot(x,(participant_cell_array{sub_id,cond}(:,21)));
            title({'Hip Joint Power','Stance Phase'});
            ylabel('Joint Moment (W/kg)');
            xlabel('Percent Stance (%)');

            %%KNEE%%
            %Knee flexion angle stance
            hold on;
            subplot(3,4,5);
            plot(x,(participant_cell_array{sub_id,cond}(:,4)));
            title({'Knee Flexion Angle','Stance Phase'});
            ylabel('Joint angle (degrees)');
            xlabel('Percent Stance (%)');

            %Knee flexion angle GC
            hold on;
            subplot(3,4,6);
            plot(x,(participant_cell_array{sub_id,cond}(:,10)));
            title({'Knee Flexion Angle','Gait Cycle'});
            ylabel('Joint angle (degrees)');
            xlabel('Percent Gait Cycle (%)');

            %Knee flexion moment
            hold on;
            subplot(3,4,7);
            plot(x,(participant_cell_array{sub_id,cond}(:,17)));
            title({'Knee Joint Moment','Stance Phase'});
            ylabel('Joint Moment (%BW*Height)');
            xlabel('Percent Stance (%)');

            %Knee Joint Power
            hold on;
            subplot(3,4,8);
            plot(x,(participant_cell_array{sub_id,cond}(:,22)));
            title({'Knee Joint Power','Stance Phase'});
            ylabel('Joint Moment (W/kg)');
            xlabel('Percent Stance (%)');

            %%ANKLE%%
            %Ankle flexion angle stance
            hold on;
            subplot(3,4,9);
            plot(x,(participant_cell_array{sub_id,cond}(:,7)));
            title({'Ankle Flexion Angle','Stance Phase'});
            ylabel('Joint angle (degrees)');
            xlabel('Percent Stance (%)');

            %Ankle flexion angle GC
            hold on;
            subplot(3,4,10);
            plot(x,(participant_cell_array{sub_id,cond}(:,11)));
            title({'Ankle Flexion Angle','Gait Cycle'});
            ylabel('Joint angle (degrees)');
            xlabel('Percent Gait Cycle (%)');

            %Ankle flexion moment
            hold on;
            subplot(3,4,11);
            plot(x,(participant_cell_array{sub_id,cond}(:,20)));
            title({'Ankle Joint Moment','Stance Phase'});
            ylabel('Joint Moment (%BW*Height)');
            xlabel('Percent Stance (%)');

            %Ankle Joint Power
            hold on;
            subplot(3,4,12);
            plot(x,(participant_cell_array{sub_id,cond}(:,23)));        
            title({'Ankle Joint Power','Stance Phase'});
            ylabel('Joint Moment (W/kg)');
            xlabel('Percent Stance (%)');
            
            sgtitle(participant_cell_array{1,cond}, 'Interpreter', 'none', 'FontSize', 24) 

        end

        labels = {};
        for l = 1:length(current_struct)
            name = fliplr(current_struct(l).name);
            labels{l} = ([fliplr(name(5:10))]);
        end

        legend(labels, 'Interpreter', 'none');

    end
end



